components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -392,8 +392,29 @@ public static String getCallbackURL(org.wso2.carbon.identity.recovery.model.Prop
                 break;
             }
         }
-        URI uri = new URI(callbackURL);
-        return new URI(uri.getScheme(), uri.getAuthority(), uri.getPath(), null, null).toString();
+
+        if (StringUtils.isNotBlank(callbackURL)) {
+            URI uri = new URI(callbackURL);
+            callbackURL = new URI(uri.getScheme(), uri.getAuthority(), uri.getPath(), null, null)
+                    .toString();
+        }
+        return callbackURL;
+    }
+
+    /**
+     * Extracts the boolean value of 'isUserPortalURL' from the properties.
+     *
+     * @param properties from the request
+     * @return the boolean value of 'isUserPortalURL'
+     */
+    public static boolean isUserPortalURL(org.wso2.carbon.identity.recovery.model.Property[] properties) {
+
+        for (org.wso2.carbon.identity.recovery.model.Property property : properties) {
```

Comment:
shall we add a null check on properties?

Reply:
fixed